### PR TITLE
build: drop the use of XC_AMEND_DISTCLEAN

### PR DESCRIPTION
**Pull request by jengelh**
Because automake used to delete depdirs at once (.deps) and there was an issue with portability, curl's XC_AMEND_DISTCLEAN greps the Makefiles in an attempt to build a list of all depfiles and delete them individually instead.

Since commit 08849db866b44510f6b8fd49e313c91a43a3dfd3, automake switched from deleting directories to individual files. curl's custom logic now finds a lot more results with the grep (the filtering of these results isn't great), which causes a massive bloating of the Makefile in the order of O(n^2).

References: https://github.com/curl/curl/issues/9843
References: https://debbugs.gnu.org/cgi/bugreport.cgi?bug=59288

### CODE REVIEW

**Comment by ilmari-lauhakangas** [comment]
I confirm this fixes the issue. Maximum respect for sniffing this out!

**Review by bagder — APPROVED** [review]
Fantastic!

File: configure.ac [hunk 1]
```diff
@@ -4665,8 +4665,6 @@ AC_OUTPUT
 
 CURL_GENERATE_CONFIGUREHELP_PM
 
-XC_AMEND_DISTCLEAN([lib src tests/unit tests/server tests/libtest docs/examples])
-
```
[the remark this answers is not in the document]
Comment (jengelh) [reply configure.ac:4669]: Does even *any* part of that xc stuff still have a reason to exist?

**Comment by jengelh** [comment]
extra commit added

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**